src/storage/mod.rs

```diff
@@ -919,6 +919,60 @@ impl Storage {
             .flatten()
     }
 
+    pub fn async_raw_batch_get(
+        &self,
+        ctx: Context,
+        keys: Vec<Vec<u8>>,
+    ) -> impl Future<Item = Vec<Result<KvPair>>, Error = Error> {
+        const CMD: &str = "raw_batch_get";
+        let engine = self.get_engine();
+        let priority = readpool::Priority::from(ctx.get_priority());
+
+        let keys: Vec<Key> = keys.into_iter().map(Key::from_encoded).collect();
+
+        let res = self.read_pool.future_execute(priority, move |ctxd| {
+            let mut _timer = {
+                let ctxd = ctxd.clone();
+                let mut thread_ctx = ctxd.current_thread_context_mut();
+                thread_ctx.start_command_duration_timer(CMD, priority, true)
+            };
+
+            Self::async_snapshot(engine, &ctx)
+                .and_then(move |snapshot: Box<Snapshot>| {
+                    let mut thread_ctx = ctxd.current_thread_context_mut();
+                    let _t_process = thread_ctx.start_processing_read_duration_timer(CMD);
+
+                    // no scan_count for this kind of op.
+                    let mut stats = Statistics::default();
+                    thread_ctx.collect_key_reads(CMD, keys.len() as u64);
```

Comment:
If I remembered correctly, `collect_key_reads` should collect the count of successful read keys. You can directly use `stats.data.flow_stats.read_keys` as the parameter value for `collect_key_reads`.